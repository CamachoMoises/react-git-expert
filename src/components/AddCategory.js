import React, { useState } from 'react';

export const AddCategory = () => {
    const [inputValue, setInputValue] = useState('New category')

    const handleInputChange=(e)=>{
        setInputValue(e.target.value)
    }
    const handleSubmit= (e)=>{
        e.preventDefault()
        console.log('submitted');
    }   
	return (
		<form onSubmit={handleSubmit}>
			<h2> Add Category {inputValue} </h2> 
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />

		</form>
	);
};
