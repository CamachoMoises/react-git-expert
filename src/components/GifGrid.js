import React, { useEffect, useState } from 'react';

export const GifGrid = ({ category }) => {
	const [images, setImages] = useState([]);
	useEffect(() => {
		getGif();
	}, []);

	const getGif = async () => {
		const url = 'https://api.giphy.com/v1/gifs/search?q=Speed+Racer&limit=10&api_key=wnzsNoA0er1NfL4znwh2pMAqRWa5BTi2';
		const resp = await fetch(url);
		const { data } = await resp.json();
		const gifs = data.map((gif) => {
			return {
				id: gif.id,
				title: gif.title,
				url: gif.images?.downsized_small.mp4,
			};
		});
		console.log(gifs);
		setImages(gifs);
	};

	return (
		<div>
			<h3> {category} </h3>
			<ol>
				{images.map(({id, title}) => (
					<li key={id}> {title}</li>
				))}
			</ol>
		</div>
	);
};
