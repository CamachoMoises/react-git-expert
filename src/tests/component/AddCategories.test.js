import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe('Test in AddCategory component', () => {
	const setCategories = jest.fn();
	let wrapper = shallow(<AddCategory setCategories={setCategories} />);
	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />);
	});
	test('should show the component snapshot ', () => {
		expect(wrapper).toMatchSnapshot();
	});
	test('should change text box', () => {
		const input = wrapper.find('input');
		input.simulate('change', { target: { value: 'value' } });
		const inputValue = wrapper.find('p');
		expect(inputValue.text().trim()).toBe('value');
	});
	test("shouldn't post the information with submit", () => {
		// wrapper.find('form').simulate('submit', {preventDefault: ()=>{}});
		wrapper.find('form').simulate('submit', { preventDefault() {} });
		expect(setCategories).not.toHaveBeenCalled();
	});
    test('should call setCategories an clean text box', () => {
        const input = wrapper.find('input');
		input.simulate('change', { target: { value: 'value' } });
		wrapper.find('form').simulate('submit', { preventDefault() {} });
		expect(setCategories).toHaveBeenCalled();
		// expect(setCategories).toHaveBeenCalledTimes(2);
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
        expect( input.prop('value') ).toBe('');

    })
    

});
