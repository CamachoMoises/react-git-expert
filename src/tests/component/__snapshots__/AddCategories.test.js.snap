// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in AddCategory component should show the component snapshot  1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
     Add Category 
     
  </h2>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
